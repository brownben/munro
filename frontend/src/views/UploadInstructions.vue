<!--
  Instructions for Uploading results
-->
<template>
  <Layout title="Upload Instructions">
    <vue-headful
      :head="{
        meta: { name: 'robots', content: 'all' },
      }"
      title="Munro - Upload Instructions"
      description="Instrcutions for how to upload results to Munro, the Fast and Easy Results System for Orienteering Leagues. A simple way to calculate the results for orienteering leagues, with search and sort features"
      url="https://munro-leagues.herokuapp.com/upload/instructions"
    />
    <InstructionCard :number="1">
      <template #instruction>
        Go to
        <router-link to="/upload/file" class="link text-main-700"
          >https://munro-leagues.herokuapp.com/upload/file</router-link
        >
      </template>
      <template #body>
        <img
          src="@/assets/images/UploadInstructions-EmptyFormNew.png"
          alt="Empty Upload Form"
        />
      </template>
    </InstructionCard>

    <InstructionCard :number="2">
      <template #instruction>
        Enter the Event ID and the 20 digit Upload Key Provided
      </template>
      <template #body>
        <p>
          The event name should now be displayed. You can get the Event ID and
          Upload Key from your league coordinator or by emailing
          munro.leagues@gmail.com The Event ID will be the the form
          LeagueNameEventNameYYYY-MM-DD with all spaces removed
        </p>
        <img
          src="@/assets/images/UploadInstructions-KeysEnteredNew.png"
          alt="Upload Form with Keys Entered"
        />
      </template>
    </InstructionCard>

    <InstructionCard :number="3">
      <template #instruction>
        Select Your Results File
      </template>
      <template #body>
        <p>
          This file can be an OE CSV File or SITiming Processable CSV file with
          semi-colon or comma delimiter. Or another semicolon/ comma delimited
          CSV file with the following headings, these headings can be in any
          order:
        </p>
        <code
          >Name; Age Class; Club; Course; Time; Position; Noncompetitive;
          Status;
        </code>
      </template>
    </InstructionCard>

    <InstructionCard :number="4">
      <template #instruction>
        The `Upload File` Button Should Now Appear
      </template>
      <template #body>
        <p>
          If you are re-uploading results tick the 'Overwrite Existing Results'
          checkbox
        </p>
        <img
          src="@/assets/images/UploadInstructions-ButtonShowingNew.png"
          alt="Upload Form with Upload Button Visible"
        />
      </template>
    </InstructionCard>

    <InstructionCard :number="5">
      <template #instruction>
        Enter the URLs of the Results Uploaded on Other Websites eg. HTML
        Results/ Routegadget/ Winsplits.
      </template>
      <template #body>
        <p>
          If you forget, or need to change these URLs please contact your league
          admin or munro.leagues@gmail.com who can update them
        </p>
      </template>
    </InstructionCard>

    <InstructionCard :number="6">
      <template #instruction>
        Click Upload
      </template>
      <template #body>
        <p>
          A message should appear saying that the data has been sent. Then a
          message saying that the results upload is successful should appear and
          you will be redirected to the league page
        </p>
      </template>
    </InstructionCard>

    <div class="col-span-2 card-color">
      <p class="p-1">
        If you need any help or wish to upload results in a different format
        please contact me at:
        <span class="text-lg font-heading">munro.leagues@gmail.com</span>
      </p>
    </div>
  </Layout>
</template>

<script>
import Layout from '@/components/Layout'
import InstructionCard from '@/components/cards/InstructionCard'

export default {
  components: {
    Layout,
    InstructionCard,
  },
}
</script>
