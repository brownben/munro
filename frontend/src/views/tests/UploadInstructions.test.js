/*
  Upload InstructionsView Unit Tests
*/

import { mount } from '@vue/test-utils'
import UploadInstructions from '@/views/UploadInstructions'

test('Is a Vue Instance', () => {
  const wrapper = mount(UploadInstructions, {
    stubs: ['router-link', 'vue-headful'],
  })
  expect(wrapper.isVueInstance()).toBeTruthy()
})

test('Renders Correctly', () => {
  const wrapper = mount(UploadInstructions, {
    stubs: ['router-link', 'vue-headful'],
  })
  expect(wrapper.element).toMatchSnapshot()
})
