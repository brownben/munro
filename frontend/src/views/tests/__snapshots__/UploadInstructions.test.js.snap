// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Correctly 1`] = `
<div
  class="w-10/12 mx-auto md:w-3/4 xl:w-2/3 md:mt-2"
>
  <h1
    class="my-2 text-3xl font-bold md:text-35xl font-heading text-main-700"
  >
    
    Upload Instructions
  
  </h1>
   
  <div
    class="grid grid-cols-2 gap-8 mb-4 md:gap-6 lg:gap-8"
  >
    <vue-headful-stub
      description="Instrcutions for how to upload results to Munro, the Fast and Easy Results System for Orienteering Leagues. A simple way to calculate the results for orienteering leagues, with search and sort features"
      head="[object Object]"
      title="Munro - Upload Instructions"
      url="https://munro-leagues.herokuapp.com/upload/instructions"
    />
     
    <section
      class="col-span-2 card"
    >
      <p
        class="text-lg font-heading"
      >
        <span
          class="inline-flex items-center justify-center w-8 h-8 mr-2 font-bold text-center align-middle border-2 rounded-full border-main text-main font-heading font-lg"
        >
          
      1
    
        </span>
         
        <span
          class="align-middle"
        >
          
      Go to
      
          <router-link-stub
            class="link text-main-700"
            to="/upload/file"
          >
            https://munro-leagues.herokuapp.com/upload/file
          </router-link-stub>
        </span>
      </p>
       
      <div
        class="text-opacity-75 text-main-900"
      >
        <img
          alt="Empty Upload Form"
          src="@/assets/images/UploadInstructions-EmptyFormNew.png"
        />
      </div>
    </section>
     
    <section
      class="col-span-2 card"
    >
      <p
        class="text-lg font-heading"
      >
        <span
          class="inline-flex items-center justify-center w-8 h-8 mr-2 font-bold text-center align-middle border-2 rounded-full border-main text-main font-heading font-lg"
        >
          
      2
    
        </span>
         
        <span
          class="align-middle"
        >
          
      Enter the Event ID and the 20 digit Upload Key Provided
    
        </span>
      </p>
       
      <div
        class="text-opacity-75 text-main-900"
      >
        <p>
          
        The event name should now be displayed. You can get the Event ID and
        Upload Key from your league coordinator or by emailing
        munro.leagues@gmail.com The Event ID will be the the form
        LeagueNameEventNameYYYY-MM-DD with all spaces removed
      
        </p>
         
        <img
          alt="Upload Form with Keys Entered"
          src="@/assets/images/UploadInstructions-KeysEnteredNew.png"
        />
      </div>
    </section>
     
    <section
      class="col-span-2 card"
    >
      <p
        class="text-lg font-heading"
      >
        <span
          class="inline-flex items-center justify-center w-8 h-8 mr-2 font-bold text-center align-middle border-2 rounded-full border-main text-main font-heading font-lg"
        >
          
      3
    
        </span>
         
        <span
          class="align-middle"
        >
          
      Select Your Results File
    
        </span>
      </p>
       
      <div
        class="text-opacity-75 text-main-900"
      >
        <p>
          
        This file can be an OE CSV File or SITiming Processable CSV file with
        semi-colon or comma delimiter. Or another semicolon/ comma delimited
        CSV file with the following headings, these headings can be in any
        order:
      
        </p>
         
        <code>
          Name; Age Class; Club; Course; Time; Position; Noncompetitive;
        Status;
      
        </code>
      </div>
    </section>
     
    <section
      class="col-span-2 card"
    >
      <p
        class="text-lg font-heading"
      >
        <span
          class="inline-flex items-center justify-center w-8 h-8 mr-2 font-bold text-center align-middle border-2 rounded-full border-main text-main font-heading font-lg"
        >
          
      4
    
        </span>
         
        <span
          class="align-middle"
        >
          
      The \`Upload File\` Button Should Now Appear
    
        </span>
      </p>
       
      <div
        class="text-opacity-75 text-main-900"
      >
        <p>
          
        If you are re-uploading results tick the 'Overwrite Existing Results'
        checkbox
      
        </p>
         
        <img
          alt="Upload Form with Upload Button Visible"
          src="@/assets/images/UploadInstructions-ButtonShowingNew.png"
        />
      </div>
    </section>
     
    <section
      class="col-span-2 card"
    >
      <p
        class="text-lg font-heading"
      >
        <span
          class="inline-flex items-center justify-center w-8 h-8 mr-2 font-bold text-center align-middle border-2 rounded-full border-main text-main font-heading font-lg"
        >
          
      5
    
        </span>
         
        <span
          class="align-middle"
        >
          
      Enter the URLs of the Results Uploaded on Other Websites eg. HTML
      Results/ Routegadget/ Winsplits.
    
        </span>
      </p>
       
      <div
        class="text-opacity-75 text-main-900"
      >
        <p>
          
        If you forget, or need to change these URLs please contact your league
        admin or munro.leagues@gmail.com who can update them
      
        </p>
      </div>
    </section>
     
    <section
      class="col-span-2 card"
    >
      <p
        class="text-lg font-heading"
      >
        <span
          class="inline-flex items-center justify-center w-8 h-8 mr-2 font-bold text-center align-middle border-2 rounded-full border-main text-main font-heading font-lg"
        >
          
      6
    
        </span>
         
        <span
          class="align-middle"
        >
          
      Click Upload
    
        </span>
      </p>
       
      <div
        class="text-opacity-75 text-main-900"
      >
        <p>
          
        A message should appear saying that the data has been sent. Then a
        message saying that the results upload is successful should appear and
        you will be redirected to the league page
      
        </p>
      </div>
    </section>
     
    <div
      class="col-span-2 card-color"
    >
      <p
        class="p-1"
      >
        
      If you need any help or wish to upload results in a different format
      please contact me at:
      
        <span
          class="text-lg font-heading"
        >
          munro.leagues@gmail.com
        </span>
      </p>
    </div>
  </div>
</div>
`;
